pkg/kubelet/cm/cpumanager/cpu_manager.go

```diff
@@ -0,0 +1,52 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cpumanager
+
+import (
+	"k8s.io/api/core/v1"
+
+	runtimeapi "k8s.io/kubernetes/pkg/kubelet/apis/cri/v1alpha1/runtime"
+	"k8s.io/kubernetes/pkg/kubelet/cm/cpumanager/state"
+	"k8s.io/kubernetes/pkg/kubelet/status"
+)
+
+type ActivePodsFunc func() []*v1.Pod
+
+type runtimeService interface {
+	UpdateContainerResources(id string, resources *runtimeapi.LinuxContainerResources) error
+}
+
+type policyName string
+
+// Manager interface provides methods for Kubelet to manage pod cpus.
+type Manager interface {
+	// Start is called during Kubelet initialization.
+	Start(activePods ActivePodsFunc, podStatusProvider status.PodStatusProvider)
+
+	// AddContainer is called between container create and container start
+	// so that initial CPU affinity settings can be written through to the
+	// container runtime before the first process begins to execute.
+	AddContainer(p *v1.Pod, c *v1.Container, containerID string) error
```

Comment:
@derekwaynecarr -- as discussed in the next PR for this feature we considered changing the name of these functions from `RegisterContainer` and `UnregisterContainer` to indicate when in the container lifecycle they should be called.

Our current plan is to have an interface for internal container lifecycle callbacks within the container manager. Those functions will have names like `PreStartContainer`, `PreStopContainer`, `PostStopContainer` etc. However, we want `UnregisterContainer` to get called during both `removeContainer` and `killContainer`, so neither `PreStopContainer` nor `PostStopContainer` makes sense...

What we would like to do is call the lifecycle callback `PreStopContainer` from `killContainer` and call `PostStopContainer` from `removeContainer`. In the hooks impl we get from the container manager, both `PreStopContainer` and `PostStopContainer` will call `cpuManager.UnregisterContainer`.

As a compromise, I have renamed `RegisterContainer` => `AddContainer` and `UnregisterContainer` => `RemoveContainer`, and clarified the banner comments. Hope this works for you.